pip/cmdoptions.py

```diff
@@ -354,9 +363,53 @@ def editable():
     action='store_false',
     default=True,
     help=('Do not Find and prefer wheel archives when searching indexes and '
-          'find-links locations.'),
+          'find-links locations. DEPRECATED in favour of --no-binary.'),
 )
 
+
+def _get_format_control(values, option):
+    """Get a format_control object."""
+    return getattr(values, option.dest)
+
+
+def _handle_no_binary(option, opt_str, value, parser):
+    existing = getattr(parser.values, option.dest)
+    fmt_ctl_handle_mutual_exclude(
+        value, existing.no_binary, existing.only_binary)
+
+
+def _handle_only_binary(option, opt_str, value, parser):
+    existing = getattr(parser.values, option.dest)
+    fmt_ctl_handle_mutual_exclude(
+        value, existing.only_binary, existing.no_binary)
+
+
+def no_binary():
+    return Option(
+        "--no-binary", dest="format_control", action="callback",
```

Comment:
I don't like "binary".  I think it should be "wheel".   Are we considering other built package types?

In the Packaging User Guide, we specifically describe wheels generally as "built" packages.   "Binary" is reserved for when wheels contain compiled extensions.